packages/mana/accessbasevector.go

```diff
@@ -95,10 +95,16 @@ func (a *AccessBaseManaVector) GetMana(nodeID identity.ID, t ...time.Time) (floa
 func (a *AccessBaseManaVector) GetManaMap(timestamp ...time.Time) (res NodeMap, t time.Time, err error) {
 	a.Lock()
 	defer a.Unlock()
+	updateAt := time.Now()
+	if len(timestamp) > 0 {
+		if timestamp[0].Before(t) {
+			t = timestamp[0]
+		}
+	}
 	res = make(map[identity.ID]float64)
 	for ID := range a.vector {
 		var mana float64
-		mana, t, err = a.getMana(ID, timestamp...)
+		mana, t, err = a.getMana(ID, updateAt)
```

Comment:
probably we don't need `t` here, or just check that the returned time is the same as `t`. Why update the returned `t` for every getMana call?